Code/GraphMol/SmilesParse/CXSmilesOps.cpp

```diff
@@ -2371,15 +2371,75 @@ void appendToCXExtension(const std::string &addition, std::string &base) {
 }
 
 }  // namespace
+
+void logLNandSgH(const ROMol &mol) {
+  std::string lns;
+  if (mol.getPropIfPresent(common_properties::molFileLinkNodes, lns)) {
+    BOOST_LOG(rdWarningLog) << "mol has link nodes which are not currently supported" << std::endl;
+  }
+  const auto &sgs = getSubstanceGroups(mol);
+  auto parent_check = std::any_of(sgs.cbegin(), sgs.cend(), [&](const SubstanceGroup &sg ) {
+    if (sg.hasProp("PARENT")) {
+      return true;
+    }
+    return false;
+  });
+  if (parent_check) {
+    BOOST_LOG(rdWarningLog) << "Substance group hierarchy is not always preserved." << std::endl;
+  }
+}
+
+std::string getCXExtensions(const MOL_SPTR_VECT &mols, std::uint32_t flags) {
+    for (const auto& mol : mols) {
+        logLNandSgH(*mol);
+        if (!mol->hasProp(RDKit::common_properties::_smilesAtomOutputOrder) ||
+          !mol->hasProp(RDKit::common_properties::_smilesBondOutputOrder)) {
+        throw ValueErrorException("Input molecule does not have the required "
+                                  "smiles ordering properties set");
+      }
+    }
+    RDKit::RWMol rwmol;
+
+    std::vector<unsigned int> atomOrdering;
+    std::vector<unsigned int> bondOrdering;
+
+    for (const auto& mol : mols) {
+      const auto at_count = rwmol.getNumAtoms();
+      const auto bond_count = rwmol.getNumBonds();
+
+      std::vector<unsigned int> prevAtomOrdering;
+      std::vector<unsigned int> prevBondOrdering;
+
+      rwmol.insertMol(*mol);
+
+      mol->getProp(RDKit::common_properties::_smilesAtomOutputOrder, prevAtomOrdering);
+      mol->getProp(RDKit::common_properties::_smilesBondOutputOrder, prevBondOrdering);
+      for (auto i : prevAtomOrdering) {
+        atomOrdering.push_back(i + at_count);
+      }
+      for (auto i : prevBondOrdering) {
+        bondOrdering.push_back(i + bond_count);
+      }
+    }
+
+    rwmol.setProp(RDKit::common_properties::_smilesAtomOutputOrder, atomOrdering, true); 
+    rwmol.setProp(RDKit::common_properties::_smilesBondOutputOrder, bondOrdering, true);
+
+
+    // call getCXExtensions
```

Comment:
I don't think this comment is necessary :) 

Also, there is a lot of extra lines in the code above. I think we. can remove some of that